src/events.rs

```diff
@@ -161,165 +165,189 @@ impl<T: Config> EventsDecoder<T> {
         input: &mut &[u8],
         output: &mut Vec<u8>,
     ) -> Result<(), BasicError> {
-        let ty = self
-            .metadata
-            .resolve_type(type_id)
-            .ok_or(MetadataError::TypeNotFound(type_id))?;
-
-        fn decode_raw<T: Codec>(
-            input: &mut &[u8],
-            output: &mut Vec<u8>,
-        ) -> Result<(), BasicError> {
-            let decoded = T::decode(input)?;
-            decoded.encode_to(output);
+        let all_bytes = *input;
+        // consume some bytes, moving the cursor forward:
+        decode_and_consume_type(type_id, &self.metadata.runtime_metadata().types, input)?;
+        // count how many bytes were consumed based on remaining length:
+        let consumed_len = all_bytes.len() - input.len();
+        // move those consumed bytes to the output vec unaltered:
+        output.extend(&all_bytes[0..consumed_len]);
+        Ok(())
+    }
+}
+
+// Given a type Id and a type registry, attempt to consume the bytes
+// corresponding to that type from our input.
+fn decode_and_consume_type(
```

Comment:
TODO: If we are happy and merge this, we could create a new "good first issue" to review the tests, perhaps adding some like the bitvec ones to test this method more specifically?

Reply:
Yes this would be a good method to unit test